homeassistant/components/squeezebox/browse_media.py

```diff
@@ -70,25 +71,37 @@ async def build_item_response(entity, player, payload):
 
     media_class = CONTENT_TYPE_MEDIA_CLASS[search_type]
 
-    if search_id and search_id != search_type:
-        browse_id = (SQUEEZEBOX_ID_BY_TYPE[search_type], search_id)
+    if "*" in search_id:
```

Comment:
I don't like that integrations have to look for `*` to see if it's a search query. It breaks artists that have a `*` too. It's started from a separate dialog and doesn't have to re-use the same API. 

The other tricky part is, how should search behave. For Spotify it's "global", while if you're browsing a local folder, you might be able to search through only the content of the current + subfolders.

Reply:
made a change to the search test.  Could still potentially break.  open to suggestions.  

This was an relatively effective simple approach without having to make major changes to the architecture.

Reply:
This approach is not going to work and it will break certain normal cases. We shouldn't do it like this.